tools/setup_helpers/cmake.py

```diff
@@ -338,19 +338,3 @@ def build(self, my_env):
         else:
             build_args += ['--', '-j', max_jobs]
         self.run(build_args, my_env)
-
-        # in cmake, .cu compilation involves generating certain intermediates
-        # such as .cu.o and .cu.depend, and these intermediates finally get compiled
-        # into the final .so.
-        # Ninja updates build.ninja's timestamp after all dependent files have been built,
-        # and re-kicks cmake on incremental builds if any of the dependent files
-        # have a timestamp newer than build.ninja's timestamp.
-        # There is a cmake bug with the Ninja backend, where the .cu.depend files
-        # are still compiling by the time the build.ninja timestamp is updated,
-        # so the .cu.depend file's newer timestamp is screwing with ninja's incremental
-        # build detector.
-        # This line works around that bug by manually updating the build.ninja timestamp
-        # after the entire build is finished.
-        ninja_build_file = os.path.join(self.build_dir, 'build.ninja')
-        if os.path.exists(ninja_build_file):
-            os.utime(ninja_build_file, None)
```

Comment:
Things might have been different when this comment was first made. But now I believe the `.depend` files are generated after the `build.ninja` file by design. Yes, it causes the `build.ninja` file to be regenerated, as well as numerous `.cmake` and `.pre-gen` files. However, this, regeneration takes only few seconds, and does not in itself trigger other building steps.

Touching `build.ninja` suppresses regeneration of build.ninja. As a result. it masks triggers for rebuild. I have so far identified three bogus triggers, patched in this PR. The problem is, touching `build.ninja` also masks legitimate triggers. That's what was causing the issue this PR is attempting to solve.